Userland/Libraries/LibWeb/Painting/DisplayListPlayerSkia.cpp

```diff
@@ -909,9 +911,38 @@ CommandResult DisplayListPlayerSkia::draw_line(DrawLine const& command)
     auto from = to_skia_point(command.from);
     auto to = to_skia_point(command.to);
     auto& canvas = surface().canvas();
+
     SkPaint paint;
     paint.setStrokeWidth(command.thickness);
     paint.setColor(to_skia_color(command.color));
+
+    switch (command.style) {
+    case Gfx::LineStyle::Solid:
+        break;
+    case Gfx::LineStyle::Dotted: {
+        auto length = command.to.distance_from(command.from) - 1;
```

Comment:
I am a bit confused why 1 need to be subtracted from distance. Could you add a little comment with explanation why it's needed, please? Also, then we should probably verify distance between points is more then zero, so we won't end up with negative length.

Reply:
Thanks for reviewing. I've added a comment explaining why 1 needs to be subtracted. I also now check the length is larger then 0 before appling the effect.

Reply:
Actually, thinking about it more. A better solution would be to extend the line by less then an interval. That way the position of the last dot won't be off by one. I've implemented that fix